File: notify/v1/notification/create-send-notification-with-badge/create-send-notification-with-badge.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Notify.V1.Service;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var apn = new Dictionary<string, Object>()
        {
            {"aps", new Dictionary<string, Object>()
                {
                    {"alert", new Dictionary<string, Object>()
                        {
                            {"title", "Bob alert"},
                            {"body", "Bob, you just received a badge"}
                        }},
                    {"badge", 1}
                }}
        };

        var identity = new List<string> {
            "00000001"
        };

        var notification = NotificationResource.Create(
            apn: apn,
            identity: identity,
            pathServiceSid: "ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        Console.WriteLine(notification.Sid);
    }
}

File: notify/v1/notification/create-send-notification-with-badge/create-send-notification-with-badge.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

notification = @client.notify
                      .services('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
                      .notifications
                      .create(apn: {
                           'aps' => {
                               'alert' => {
                                   'title' => 'Bob alert',
                                   'body' => 'Bob, you just received a badge'
                               },
                               'badge' => 1
                           }
                       }, identity: '00000001')

puts notification.sid
